### ISSUE TITLE
hybrid SBT/LCA "reverse index" databases?

### ISSUE THREAD

**Issue by ctb** [sourmash-bio/sourmash#943]
relevant to #756 and #925, perhaps.

It might be interesting to consider a mixture of SBTs (at high level) and LCA/revindex databases (at low level). The rationale is that LCA databases are a lot faster than SBTs, but less scalable; that (lack of) scalability is partly because revindex databases use a single JSON file to hold all of the hashes for a bunch of signatures. Because of this, it seems like it might be efficient to use them for collections of genomes that share a lot of hashes.

So, basically, you could imagine having SBTs that, at a high level, cluster subtrees by shared hashes; and then at lower levels, switch to putting localized collections of signatures into LCA databases.

Juuuuust a thought...

**Comment by ctb** [comment]
all that is new is old #418

**Comment by ctb** [comment]
there's a related idea that comes to mind, as I start to think about "low rank" k-mers (genus/species specific). if we take an ANI-based taxonomy like GTDB across all their genomes of interest (not just 25k!), and eliminate all k-mers that belong to more than one genus, can we provide an index that lets us super rapidly identify genus and jump to a genus-level database?

I'm thinking SBT with "leaf" nodes being these genus-level things, perhaps as bloom filters; and then pointing at LCA databases. OR, vice versa, maybe revindex/LCA at top, with SBTs for the genus-level stuff. Or a new data structure, revindex/LCA at top, pointing to SBTs for genus level stuff.

**Comment by ctb** [comment]
closing as obsolete - the conversation has moved onto much faster mastiff-style databases :)
